src/NzbDrone.Core/Music/ArtistRepository.cs

```diff
@@ -42,5 +48,39 @@ public Artist FindByName(string cleanName)
             return Query.Where(s => s.CleanName == cleanName)
```

Comment:
Not your code, but this could be simplified to `return Query.SingleOrDefault(s => s.CleanName == cleanName);` while you are in here.

Reply:
Not sure this is a good idea - I don't think the data mapper knows how to translate `Query.SingleOrDefault(s => s.CleanName == cleanName)` into a `WHERE` query.

Testing quickly, this is the query you get when you use `Query.Where(s => s.CleanName == cleanName).SingleOrDefault()`:

```
==== Begin Query Trace ====

QUERY TYPE:
Text

QUERY TEXT:
SELECT [t0].[ForeignArtistId],[t0].[MBId],[t0].[TADBId],[t0].[DiscogsId],[t0].[AMId],[t0].[Name],[t0].[CleanName],[t0].[SortName],[t0].[Overview],[t0].[Disambiguation],[t0].[ArtistType],[t0].[Monitored],[t0].[AlbumFolder],[t0].[LastInfoSync],[t0].[LastDiskSync],[t0].[Status],[t0].[Path],[t0].[Images],[t0].[Links],[t0].[Genres],[t0].[Added],[t0].[ProfileId],[t0].[LanguageProfileId],[t0].[MetadataProfileId],[t0].[Tags],[t0].[AddOptions],[t0].[Ratings],[t0].[Members],[t0].[Id] FROM [Artists] [t0] WHERE ([t0].[CleanName] = @P0)

PARAMETERS:
@P0 = ["theblackeyedpeas"]

==== End Query Trace ====

```
Note the `WHERE` clause at the end.

If you switch to `Query.SingleOrDefault(s => s.CleanName == cleanName)` you get:
```
==== Begin Query Trace ====

QUERY TYPE:
Text

QUERY TEXT:
SELECT [t0].[ForeignArtistId],[t0].[MBId],[t0].[TADBId],[t0].[DiscogsId],[t0].[AMId],[t0].[Name],[t0].[CleanName],[t0].[SortName],[t0].[Overview],[t0].[Disambiguation],[t0].[ArtistType],[t0].[Monitored],[t0].[AlbumFolder],[t0].[LastInfoSync],[t0].[LastDiskSync],[t0].[Status],[t0].[Path],[t0].[Images],[t0].[Links],[t0].[Genres],[t0].[Added],[t0].[ProfileId],[t0].[LanguageProfileId],[t0].[MetadataProfileId],[t0].[Tags],[t0].[AddOptions],[t0].[Ratings],[t0].[Members],[t0].[Id] FROM [Artists] [t0]

PARAMETERS:

==== End Query Trace ====
```

There's no `WHERE` clause here so it's just fetching all the artists and letting `C#` do the searching.  Presumably it's better to let the database do this so it can use the indices?

Reply:
Good to know. Sounds good.